gcloud/rest/auth/iam.py

```diff
@@ -0,0 +1,149 @@
+import json
+import threading
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Union  # pylint: disable=unused-import
+
+import requests
+
+from .token import Token
+from .token import Type
+from .utils import encode
+
+
+API_ROOT_IAM = 'https://iam.googleapis.com/v1'
+API_ROOT_IAM_CREDENTIALS = 'https://iamcredentials.googleapis.com/v1'
+SCOPES = ['https://www.googleapis.com/auth/iam']
+
+
+class IamClient:
+    def __init__(self,
+                 service_file=None,  # type: Optional[str]
+                 session=None,  # type: requests.Session
```

Comment:
Add `Optional` to all of these

Reply:
so my understanding was that there won't be a need to add Optional to these based on gcloud-aio's Type annotations - https://github.com/talkiq/gcloud-aio/blob/master/auth/gcloud/aio/auth/token.py#L65.

Reply:
Looks like these are incorrect in `gcloud-aio` as well; these would only be accepted if we have the `--no-strict-optional` flag set in mypy, which we should avoid since it tends to help mask missing null check issues. We should similarly make this change in `-aio`, since it'd be great to be more accurate about these things.